botcore/monkey_patches.py

```diff
@@ -0,0 +1,78 @@
+import logging
+from datetime import datetime, timedelta
+from functools import partial, partialmethod
+
+from disnake import Forbidden, http
+from disnake.ext import commands
+
+log = logging.getLogger(__name__)
+
+
+class _Command(commands.Command):
+    """
+    A `disnake.ext.commands.Command` subclass which supports root aliases.
+
+    A `root_aliases` keyword argument is added, which is a sequence of alias names that will act as
+    top-level commands rather than being aliases of the command's group. It's stored as an attribute
+    also named `root_aliases`.
+    """
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.root_aliases = kwargs.get("root_aliases", [])
+
+        if not isinstance(self.root_aliases, (list, tuple)):
+            raise TypeError("Root aliases of a command must be a list or a tuple of strings.")
+
+
+class _Group(commands.Group):
+    """
+    A `disnake.ext.commands.Group` subclass which supports root aliases.
+
+    A `root_aliases` keyword argument is added, which is a sequence of alias names that will act as
+    top-level groups rather than being aliases of the command's group. It's stored as an attribute
+    also named `root_aliases`.
+    """
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.root_aliases = kwargs.get("root_aliases", [])
+
+        if not isinstance(self.root_aliases, (list, tuple)):
+            raise TypeError("Root aliases of a group must be a list or a tuple of strings.")
```

Comment:
Group is a subclass of Command, we can just use diamond-order-inheritance:
```suggestion
class _Group(commands.Group, _Command):
    """
    A `disnake.ext.commands.Group` subclass which also supports root aliases.

    A `root_aliases` keyword argument is added, which is a sequence of alias names that will act as
    top-level groups rather than being aliases of the command's group. It's stored as an attribute
    also named `root_aliases`.
    """
```

Reply:
Seems good, pushed to branch, thanks :D